apps/financial-aid/backend/src/app/modules/application/application.service.ts

```diff
@@ -218,6 +218,45 @@ export class ApplicationService {
     return application
   }
 
+  async findByApplicationSystemId(
+    id: string,
+  ): Promise<ApplicationModel | null> {
+    const application = await this.applicationModel.findOne({
+      where: { applicationSystemId: id },
+      include: [
+        {
+          model: ApplicationEventModel,
+          as: 'applicationEvents',
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: ApplicationFileModel,
+          as: 'files',
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: AmountModel,
+          as: 'amount',
+          include: [{ model: DeductionFactorsModel, as: 'deductionFactors' }],
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: DirectTaxPaymentModel,
+          as: 'directTaxPayments',
+        },
+      ],
+    })
```

Comment:
thinking findbyId is using the same includes, can we maybe make an one array that they both could use, so not to forget to update either 😃 
Or use the same endpoint, and have a boolean where isApplicationId?, because the endpoints are the same